### PR TITLE
Rebuild match events displays

### PR DESCRIPTION
**Pull request by kubaolszewski**

### CODE REVIEW

File: lib/app/features/home/home_tabs/upcoming_tab/cubit/upcoming_tab_cubit.dart [hunk 1]
```diff
@@ -15,9 +15,9 @@ class UpcomingTabCubit extends Cubit<UpcomingTabState> {
   final HomeScreenRepository homeScreenRepository;
 
   Future<void> fetchUpcomingMatches(DateTime date) async {
-    String formattedStringDate = DateFormat('yyyy-MM-dd').format(date);
-    DateTime formattedDate = DateFormat('yyyy-MM-dd').parse(formattedStringDate);
-    DateTime formattedActualDate = DateTime.now().formatActualDateToDateTime('yyyy-MM-dd');
+    String formattedStringDate = DateFormat(AppConstVariables.dateYyyyMmDdDashFormat).format(date);
```
Comment (krzychudz) [inline lib/app/features/home/home_tabs/upcoming_tab/cubit/upcoming_tab_cubit.dart]: Avoid using types explicitly if the value does not change later - use just final.

File: lib/app/features/home/match_details/match_details_screen.dart [hunk 2]
```diff
@@ -17,11 +17,11 @@ class MatchDetailsScreen extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    final int teamID = liveMatch.fixture?.id ?? AppConstVariables.intPlaceholder;
+    final int matchID = liveMatch.fixture?.id ?? AppConstVariables.intPlaceholder;
```
Comment (krzychudz) [inline lib/app/features/home/match_details/match_details_screen.dart]: Same here - looks like there is no need to write int, you can use just final.

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/first_half_events_display.dart [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        primary: false,
+        children: [
+          Row(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              Expanded(
+                child: Column(
+                  children: [
+                    for (final event in matchEvents) ...[
+                      if (event.team?.id != null && event.time?.elapsed != null && event.player?.name != null)
```
Comment (krzychudz) [inline lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/first_half_events_display.dart]: I would write an extension to event list to get only the valid ones -> so filter them by `event.team?.id != null && event.time?.elapsed != null && event.player?.name != null` - and use here as `matchEvents.validEvents` or something similar.

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/first_half_events_display.dart [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                          const SizedBox.shrink()
+                    ]
+                  ],
+                ),
+              ),
+              Expanded(
+                child: Column(
+                  children: [
+                    for (final event in matchEvents) ...[
+                      if (event.team?.id != null && event.time?.elapsed != null && event.player?.name != null)
```
Comment (krzychudz) [inline lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/first_half_events_display.dart]: Same here

File: lib/app/features/home/home_tabs/upcoming_tab/cubit/upcoming_tab_cubit.dart (hunk 1 above)
Comment (kubaolszewski) [reply lib/app/features/home/home_tabs/upcoming_tab/cubit/upcoming_tab_cubit.dart]: done

File: lib/app/features/home/match_details/match_details_screen.dart (hunk 2 above)
Comment (kubaolszewski) [reply lib/app/features/home/match_details/match_details_screen.dart]: done

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/first_half_events_display.dart (hunk 3 above)
Comment (kubaolszewski) [reply lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/first_half_events_display.dart]: right, done

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/first_half_events_display.dart (hunk 4 above)
Comment (kubaolszewski) [reply lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/first_half_events_display.dart]: done

**Review by krzychudz — APPROVED** [review]

**Pull request merged**